### ISSUE TITLE
Dirty fragment does not have attributes updated by server response

### ISSUE THREAD

**Issue by patocallaghan** [adopted-ember-addons/ember-data-model-fragments#418]
If you have a dirty fragment where calling `changedAttributes()` gives a list of properties, any server response will not update any of those properties listed and leave them as-is. This is especially noticeable if any of the attributes in the response are different than those on the local fragment. This is different to how attributes work for Ember Data models, there it seems responses will always override local attributes.

Take the following example where `person` is an Ember Data model. We can see that a server response that is different than the current local value will always override it.

```js
console.log(person.title); // Miss
person.set('title', 'Ms.');
console.log(person.title); // Ms.
console.log(person.changedAttributes()) // { title: ['Miss', 'Ms.']}

await person.save();
// Server returns `title: 'Mrs.'` in the response

console.log(person.title) // Mrs.
```

This is not the same for model fragments. Now take the following where on `person` we have a fragment `name` which has attrs `first` and `last`

```js
console.log(person.name.serialize()); // { first: 'Jon', last: 'Snow' }
person.name.set('first', 'Robert');
console.log(person.name.serialize()); // { first: 'Robert', last: 'Snow' }
console.log(person.name.changedAttributes()) // { first: ['Jon', 'Robert']}

await person.save();
// Server returns `name: { first: 'Richard', last: 'Snow' }` in the response

console.log(person.name.serialize()) // { first: 'Robert', last: 'Snow' }
// `person.name.first` has stayed as `Robert` instead of changing to `Richard`
```

This worked pre-5.0.0-beta. I'll dig in a little and see what I can find.

**Comment by patocallaghan** [comment]
This looks to be related to the fix in https://github.com/adopted-ember-addons/ember-data-model-fragments/pull/404

**Comment by patocallaghan** [comment]
Fixed in #404
